[plugin-web-app] Align syntax of element steps (#4219)
vividus-framework/vividus: docs/modules/plugins/pages/plugin-web-app.adoc

[source,gherkin]
----
When I change context to element located by `id(rainbow)`
Then context element has CSS property `color` with value that is equal to `rgba(0, 0, 0, 1)`
----

=== Compare sizes of the elements

Check that all elements found on the page by the given locator have the same dimension: width / height.

NOTE: Element's dimension is measured in *pixels* and means the size that given element occupies on the web-page
in a browser. Element size may vary in different browsers, it also depends on screen resolution, page scaling,
scripts running on the page.

[source,gherkin]
----
Then each element located by `$locator` has same `$dimension`
----

_Deprecated syntax (will be removed in VIVIDUS 0.7.0)_:
[source,gherkin]
----
Then each element located `$locator` has same `$dimension`
----

* `$locator` - The <<_locator,locator>> to find an elements to compare.
* `$dimension` - The elements dimension to compare. Possible values: _width_, _height_

.Check that all menu items have the same size (width and height)
[source,gherkin]
----
Then each element located `id(context-menu-item)` has same `width`
Then each element located `id(context-menu-item)` has same `height`
----

> === Check element width in a percentage

Reviewer: in this case the width is calculated relatively to body, please add this in docs

Then `${qrCodeLink}` is equal to `https://www.example.com`
----

=== Element Steps

=== Upload file

> This step uploads a file with the given relative path.
> 
> NOTE: A relative path starts from some given working directory, avoiding the need to provide the full absolute path (i.e. 'about.jpeg' for the files placed in the src/main/resources directory or '/story/uploadfiles/about.png')

Reviewer 2: ```suggestion
Uploads the {xref:ROOT:glossary.adoc#_resource}[resource] or file via web interface.
```

[source,gherkin]
----
When I select element located by `$locator` and upload `$resourceNameOrFilePath`
----

_Deprecated syntax (will be removed in VIVIDUS 0.7.0)_:
[source,gherkin]
----
When I select element located `$locator` and upload file `$filePath`
----

> * `$locator` - The <<_locator,locator>> to find an element.

Reviewer 2: we need to mention it's required to specify locator for `input` element with attribute `type=file`

* `$resourceNameOrFilePath` - relative path to the file to be uploaded

.Upload file_for_upload.png file
[source,gherkin]
----

> When I select element located `By.id(uploadfile)` and upload file `/folder/file_for_upload.png`

Reviewer 2: ```suggestion
When I select element located by `id(uploadfile)` and upload file `/folder/file_for_upload.png`
```

----

=== Verify page is not refreshed after click on element

> This step clicks on the element and checks that the page has not been refreshed after clicking.

Reviewer 2: ```suggestion
Clicks on the element and checks that the page has not been refreshed after the click.
```

[source,gherkin]
----
When I click on element located by `$locator` then page does not refresh
----

_Deprecated syntax (will be removed in VIVIDUS 0.7.0)_:
[source,gherkin]
----
When I click on an element '$locator' then the page does not refresh
----

* `$locator` - The <<_locator,locator>> to find an element to click.

.Step example
[source,gherkin]
----
When I click on element located by `id(send)` then page does not refresh
----

> === Verify page is not refreshed after click on element

Reviewer 2: duplicate section?

This step clicks on the element and checks that page has not been refreshed after click.

[source,gherkin]
----
When I click on element located by `$locator` then page does not refresh
----

_Deprecated syntax (will be removed in VIVIDUS 0.7.0)_:
[source,gherkin]
----
When I click on an element '$locator' then the page does not refresh
----

* `$locator` - The <<_locator,locator>> to find an element to click.

.Step example
[source,gherkin]
----
When I click on element located by `id(send)` then page does not refresh
----

=== Right-click on the element

> This step performs right-click on the element by locator.

Reviewer 2: ```suggestion
Performs right-click on the element.
```

[source,gherkin]
----

> When I perform right-click on element located by `$locator`

Reviewer 2: ```suggestion
When I perform right click on element located by `$locator`
```

Author: 'right-click' is a common term according to https://dictionary.cambridge.org/dictionary/english/right-click

----

_Deprecated syntax (will be removed in VIVIDUS 0.7.0)_:
[source,gherkin]
----
When I perform right click on element located `$locator`
----

* `$locator` - The <<_locator,locator>> to find an element to click.

.Step example
[source,gherkin]
----
When I perform right-click on element located by `id(clickme)`
----

> === Right-click on the element

Reviewer 2: one more duplicate

This step performs right-click on the element by locator.

[source,gherkin]
----
When I perform right-click on element located by `$locator`
----

_Deprecated syntax (will be removed in VIVIDUS 0.7.0)_:
[source,gherkin]
----
When I perform right click on element located `$locator`
----

* `$locator` - The <<_locator,locator>> to find an element to click.

.Step example
[source,gherkin]
----
When I perform right-click on element located by `id(clickme)`
----

=== Hover mouse over the element

> This step moves a cursor on the element specified by the locator.

Reviewer 2: ```suggestion
Moves a mouse cursor on the element.
```

Reviewer 2: the same correction is needed for all other similar occurrences

[source,gherkin]
----
When I hover mouse over element located by `$locator`
----

_Deprecated syntax (will be removed in VIVIDUS 0.7.0)_:
[source,gherkin]
----
When I hover mouse over element located `$locator`
----

* `$locator` - The <<_locator,locator>> to an element to hover.

.Step example
[source,gherkin]
----
When I hover mouse over element located by `id(hoverme)`
----

> === Hover mouse over the element

Reviewer 2: again duplicate

This step moves a cursor on the element specified by the locator.

[source,gherkin]
----
When I hover mouse over element located by `$locator`
----

_Deprecated syntax (will be removed in VIVIDUS 0.7.0)_:
[source,gherkin]
----
When I hover mouse over element located `$locator`
----

* `$locator` - The <<_locator,locator>> to an element to hover.

.Step example
[source,gherkin]
----
When I hover mouse over element located by `id(hoverme)`
----

=== Check each element has a specific number of children elements

Checks that each element specified by the locator contains an exact number of visible child elements specified by another locator

[source,gherkin]
----
Then each element with locator `$elementLocator` has `$number` child elements with locator `$childLocator`
----

* `$elementLocator` - The <<_locator,locator>> to the parent element.
* `$number` - The expected number of children elements
* `$childLocator` - The <<_locator,locator>> to the child elements.

.Step example
[source,gherkin]
----
Then each element with locator `id(menu)` has `5` child elements with locator `id(menu-item)`
----

=== Check an element CSS property

> Check that the context element has an expected CSS property.

Reviewer 2: ```suggestion
Checks the context element has the expected CSS property.
```

> The context can be set by the <<_change_context,corresponding steps>>. If no context is set, the text will be searched on the whole page.

Reviewer 2: we do not search text in this step

in a browser. Element size may vary in different browsers, it also depends on screen resolution, page scaling,
scripts running on the page.

[source,gherkin]
----
Then each element located by `$locator` has same `$dimension`
----

_Deprecated syntax (will be removed in VIVIDUS 0.7.0)_:
[source,gherkin]
----
Then each element located `$locator` has same `$dimension`
----

* `$locator` - The <<_locator,locator>> to find an elements to compare.
* `$dimension` - The elements dimension to compare. Possible values: _width_, _height_

.Check that all menu items have the same size (width and height)
[source,gherkin]
----
Then each element located `id(context-menu-item)` has same `width`
Then each element located `id(context-menu-item)` has same `height`
----

=== Check element width in a percentage

Checks that the context element has an expected *width in a percentage* (from the style attribute)

The context can be set by the <<_change_context,corresponding steps>>. If no context is set, the text will be searched on the whole page.

[source,gherkin]
----
Then context element has width of $widthPercentage%
----

_Deprecated syntax (will be removed in VIVIDUS 0.7.0)_:
[source,gherkin]
----

> Then the context has a width of '$widthInPerc'%
> ----
> 
> * `$widthInPerc` - An expected element width in a percentage.

Reviewer 2: ```suggestion
Then the context has a width of '$widthPercentage'%
----

* `$widthPercentage ` - An expected element width in a percentage.
```

.Check element width
[source,gherkin]
----
When I change context to element located by `id(menu)`
Then context element has width of 30%
----

=== Check element width in a percentage relative to the parent element

Checks that the context element has an expected *width in a percentage* (from the style attribute) relative to the parent element

> The context can be set by the <<_change_context,corresponding steps>>. If no context is set, the text will be searched on the whole page.

Reviewer 2: no text search in this step

[source,gherkin]
----
Then context element has width of $widthPercentage% relative to parent element
----

_Deprecated syntax (will be removed in VIVIDUS 0.7.0)_:
[source,gherkin]
----

> Then the context element has a width of '$widthInPerc'% relative to the parent element
> ----
> 
> * `$widthInPerc` - An expected element width in a percentage.

Reviewer 2: ```suggestion
Then the context element has a width of '$widthPercentage'% relative to the parent element
----

* `$widthPercentage ` - An expected element width in a percentage.
```
